dev/com.ibm.ws.concurrent.persistent_fat/test-applications/schedtest/src/web/SchedulerFATServlet.java

```diff
@@ -3779,13 +3794,21 @@ public void testTwoTasksInOneTransaction(PrintWriter out) throws Exception {
             tran.commit();
         }
 
-        statusA = pollForResult(statusA.getTaskId(), 1);
-        if (statusA == null || !statusA.isDone())
-            throw new Exception("TaskA not completed in allotted interval. Status: " + statusA);
+        for (long start = System.nanoTime(); System.nanoTime() - start < TIMEOUT_NS
+                && (statusA = scheduler.getStatus(statusA.getTaskId())) != null
+                && !(statusA.hasResult() && Integer.valueOf(1).equals(statusA.getResult())); )
+            Thread.sleep(POLL_INTERVAL);
+
+        if (statusA == null || !statusA.isDone() || !Integer.valueOf(1).equals(statusA.getResult()))
+            throw new Exception("TaskA not completed with expected result in allotted interval. Status: " + statusA);
+
+        for (long start = System.nanoTime(); System.nanoTime() - start < TIMEOUT_NS
+                && (statusB = scheduler.getStatus(statusB.getTaskId())) != null
+                && !(statusB.hasResult() && Integer.valueOf(1).equals(statusB.getResult())); )
+            Thread.sleep(POLL_INTERVAL);
 
-        statusB = pollForResult(statusB.getTaskId(), 1);
-        if (statusB == null || !statusB.isDone())
-            throw new Exception("TaskB not completed in allotted interval. Status: " + statusB);
+        if (statusB == null || !statusB.isDone() || !Integer.valueOf(1).equals(statusB.getResult()))
```

Comment:
Is statusB.isDone() the same thing as statusB.hasResult()?

Reply:
They are similar, but not the same.  For repeating tasks (which doesn't apply here) a task can have multiple results - one for each execution.  The hasResult method tells you if at least one execution of the task has happened, such that you can get a current result.  The isDone method tells you that all executions have completed.  For one-time tasks, such as is the case here, they are both telling you the same thing.